bundles/org.eclipse.passage.lbc.api/src/org/eclipse/passage/lbc/internal/api/BackendLicenseLock.java

```diff
@@ -22,12 +23,12 @@
 public interface BackendLicenseLock {
 
 	// Returns simply a boolean value of license.taken < license.capacity
-	ServiceInvocationResult<Boolean> canTake(Supplier<String> condition);
+	ServiceInvocationResult<Boolean> canTake(Supplier<Optional<String>> condition);
```

Comment:
not clear why we have Optional here. What should be expected as a result of `canTake(Optional.empty())?`

Reply:
We have Optional because ConditionIdentifier class (representing condition id) is an inheritor of StringNamedData that implements Supplier<Optional<String>>. We have to use this construction to prevent doubtful dependencies (lbc.api will depend on lic.base in this case)